service/defaultcomponents/defaults.go

```diff
@@ -63,6 +64,7 @@ func Components() (
 		oidcauthextension.NewFactory(),
 		pprofextension.NewFactory(),
 		zpagesextension.NewFactory(),
+		ballastextension.NewFactory(),
```

Comment:
Please split this in a separate PR, we need to discuss about deprecating the flag in that PR, also the flag is accessed by the memory limiter processor so we need a solution for that as well.

Reply:
Sure. 
> the flag is accessed by the memory limiter processor so we need a solution for that as well.

For this one, I am thinking to expose the final `ballast_size` from the extension and force the memory_limiter to use the value set from ballast extension. we can discuss it later.

```
// GetBallastSize allows the external to get the current ballast size setting
func GetBallastSize() uint64 {
	return ballastSizeBytes
}
```